packages/carbon-graphs/src/js/helpers/axis.js

```diff
@@ -514,135 +501,162 @@ const getAxesScale = (axis, scale, config) => {
     // TICKS_MAXCOUNT AND if the Y2 is visible, then utilize a default value
     // for the ticksCount. This is based on the ranges of the Y & Y2 axes.
-    else {
-      const ticksCount = getAverageTicksCount(
-        config.axis.y.domain.upperLimit
-          - config.axis.y.domain.lowerLimit,
-        config.axis.y2.domain.upperLimit
-          - config.axis.y2.domain.lowerLimit,
-      );
 
-      const yTickValues = generateYAxesTickValues(
-        config.axis.y.domain.lowerLimit,
-        config.axis.y.domain.upperLimit,
-        ticksCount,
-        config.allowCalibration,
+      let intervalCount;
+
+      // Divide the larger range value (a) by the smaller value (b) and determine if a whole number (c) is returned.
+      // If the return value is a whole number and does not have a value grter than 10, divide the smaller range value (b) by the return value (c)
+      // and determine if a whole number is returned. If a whole number is returned, use the initial return value to configure the tick count of the graph.
+      const yRange = yUpperLimit - yLowerLimit;
+      const y2Range = y2UpperLimit - y2LowerLimit;
+      const greaterRange = yRange >= y2Range ? yRange : y2Range;
+      const lowerRange = yRange <= y2Range ? yRange : y2Range;
+      const factor = greaterRange / lowerRange;
+
+      if (Number.isInteger(factor) && Number.isInteger(lowerRange / factor) && factor > 2) {
+        intervalCount = factor;
+      } else {
+      // If any part of the step above fails, determine if the two range values are both evenly divisible by
+      // any number between 3 and 9 (resulting in a whole number).
+      // If yes, choose the highest value to configure the tick count.
+        for (let _commonDenominator = 9; _commonDenominator >= 3; _commonDenominator -= 1) {
+          if (Number.isInteger(greaterRange / _commonDenominator) && Number.isInteger(lowerRange / _commonDenominator)) {
+            intervalCount = _commonDenominator;
+            break;
+          }
+        }
+      }
+
+      // Default ticksCount Behavior
+      // If both checks above fail, proceed with the current logic to configure the tick counts
+      if (utils.isUndefined(intervalCount)) {
+        intervalCount = getAverageTicksCount(yRange, y2Range) + 1;
+      }
+
+      // generating ticksCount based on the calculated value above
+      yTickValues = generateYAxesTickValues(
+        yLowerLimit,
+        yUpperLimit,
+        intervalCount - 1,
         config.axis.y,
       );
-      const y2TickValues = generateYAxesTickValues(
-        config.axis.y2.domain.lowerLimit,
-        config.axis.y2.domain.upperLimit,
-        ticksCount,
-        config.allowCalibration,
+      y2TickValues = generateYAxesTickValues(
+        y2LowerLimit,
+        y2UpperLimit,
+        intervalCount - 1,
         config.axis.y2,
       );
+    }
+    if (
+      config.axis.y.suppressTrailingZeros
+        && utils.isUndefined(config.axis.y.ticks.format)
+    ) {
```

Comment:
Some of these `if` formats are inconsistent with the rest

Reply:
Some of they were automatically updated by the linter. I can take a look at them.

Reply:
Updated here: 0b0f666a6b1567b0fe02134c22c047f146fe9ca7